flocker/node/agents/test/test_blockdevice.py

```diff
@@ -698,6 +752,186 @@ def add_application_with_volume(node_state):
                                   mountpoint=FilePath(b"/data")))})
 
 
+_BLOCKDEVICE_PREFIX = "blockdevice-"
+
+
+def _get_dataset_id_for_test_blockdevice_id(test_blockdevice_id):
+    """
+    Determine the dataset_id that was used to create a test_blockdevice_id. For
+    use with _create_test_blockdevice_id_for_dataset_id for validation of
+    opaque test blockdevice_ids with the dataset_ids they were constructed
+    from.
+
+    :param unicode test_blockdevice_id: test blockdevice_id returned by
+        _create_test_blockdevice_id_for_dataset_id.
+
+    :returns: The dataset_id used to create the test blockdevice_id.
+    """
+    return test_blockdevice_id[len(_BLOCKDEVICE_PREFIX):]
+
+
+def _create_test_blockdevice_id_for_dataset_id(dataset_id):
+    """
+    Creates a test-only blockdevice_id for a given dataset_id. Note, this is
+    only for use in tests that do not use the ``IBlockDeviceAPI`` as the test
+    blockdevice_ids that are returned will not work with any
+    ``IBlockDeviceAPI`` implementation. Tests that use the ``IBlockDeviceAPI``
+    should instead call api.list_volumes() to construct a ``BlockDeviceVolume``
+    from a dataset_id, and then get the blockdevice_id from that structure.
+
+    :param unicode dataset_id: the dataset_id to create a test blockdevice_id
+        for.
+
+    :returns: A test blockdevice_id unique to this dataset_id.
+    """
+    return "".join([_BLOCKDEVICE_PREFIX, dataset_id])
+
+
+def _create_test_blockdevice_volume_for_dataset_id(dataset_id,
```

Comment:
I would just inline this into  next function since it's only used there, and then you don't need to document it.

```
def f():
    def g(): ...
    x = g()
```